catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/dialog/utils/LegoSensorPortConfigDialogWrapper.java

```diff
@@ -0,0 +1,98 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.ui.dialog.utils;
+
+import android.support.annotation.IntDef;
+import android.support.test.espresso.ViewInteraction;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.uiespresso.util.UiTestUtils;
+import org.catrobat.catroid.uiespresso.util.wrappers.ViewInteractionWrapper;
+
+import java.lang.annotation.Retention;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+public final class LegoSensorPortConfigDialogWrapper extends ViewInteractionWrapper {
+
+	@Retention(SOURCE)
+	@IntDef({PORT_1, PORT_2, PORT_3, PORT_4})
+	public @interface FormulaEditorLegoPorts {}
+	public static final int PORT_1 = R.string.lego_port_1;
+	public static final int PORT_2 = R.string.lego_port_2;
+	public static final int PORT_3 = R.string.lego_port_3;
+	public static final int PORT_4 = R.string.lego_port_4;
+
+	@Retention(SOURCE)
+	@IntDef({NXT_NO_SENSOR, NXT_SENSOR_LIGHT_ACTIVE, NXT_SENSOR_SOUND, NXT_SENSOR_TOUCH, NXT_SENSOR_ULTRASONIC})
+	public @interface FormulaEditorNXTLegoSensors {}
+	public static final int NXT_NO_SENSOR = R.string.nxt_no_sensor;
+	public static final int NXT_SENSOR_LIGHT_ACTIVE = R.string.nxt_sensor_light_active;
+	public static final int NXT_SENSOR_SOUND = R.string.nxt_sensor_sound;
+	public static final int NXT_SENSOR_TOUCH = R.string.nxt_sensor_touch;
+	public static final int NXT_SENSOR_ULTRASONIC = R.string.nxt_sensor_ultrasonic;
+
+	private LegoSensorPortConfigDialogWrapper(ViewInteraction viewInteraction) {
+		super(viewInteraction);
+	}
+
+	public static LegoSensorPortConfigDialogWrapper onLegoSensorPortConfigDialog(@FormulaEditorNXTLegoSensors int sensorStringResource) {
+		String sensorString = UiTestUtils.getResourcesString(sensorStringResource);
+		String legoSensorConfigDialogTitle = UiTestUtils
+				.getResourcesString(R.string.lego_sensor_port_config_dialog_title, sensorString);
+		return new LegoSensorPortConfigDialogWrapper(onView(withText(legoSensorConfigDialogTitle)));
+	}
+
+	public LegoSensorPortConfigDialogWrapper performClickOnOK() {
+		onView(withText(R.string.ok)).perform(click());
+		return this;
+	}
+
+	public LegoSensorPortConfigDialogWrapper checkDialogVisible() {
+		viewInteraction.check(matches(isDisplayed()));
```

Comment:
with .inRoot(isDialog()) you could verify that the text/id youre matching is inside of a dialog:
viewInteraction.inRoot(isDialog()).check(matches(isDisplayed))